sparkle/src/components/WindowUtility.tsx

```diff
@@ -0,0 +1,60 @@
+import { useEffect, useState } from "react";
+
+// Define breakpoints
+export const breakpoints = {
+  xs: 0,
+  sm: 640,
+  md: 768,
+  lg: 1024,
+  xl: 1280,
+  xxl: 1536,
+};
+
+// Helper function to determine active breakpoint
+function getActiveBreakpoint(width: number): keyof typeof breakpoints {
+  if (width >= breakpoints.xxl) {
+    return "xxl";
+  }
+  if (width >= breakpoints.xl) {
+    return "xl";
+  }
+  if (width >= breakpoints.lg) {
+    return "lg";
+  }
+  if (width >= breakpoints.md) {
+    return "md";
+  }
+  if (width >= breakpoints.sm) {
+    return "sm";
+  }
+  return "xs";
+}
+
+// Custom hook to get window size and active breakpoint
+export function useWindowSize() {
+  const [windowSize, setWindowSize] = useState({
+    width: undefined as number | undefined,
+    height: undefined as number | undefined,
+    activeBreakpoint: "xs" as keyof typeof breakpoints,
+  });
```

Comment:
This type should be extracted in an interface so we can remove the `as`, something like this:

```suggestion

interface WindowSizeState {
  width: number | undefined;
  height: number | undefined;
  activeBreakpoint: keyof typeof breakpoints;
}

// Custom hook to get window size and active breakpoint
export function useWindowSize() {
  const [windowSize, setWindowSize] = useState<WindowSizeState>({
    width: undefined,
    height: undefined,
    activeBreakpoint: "xs"
  });
```